src/converters/app-converter.ts

```diff
@@ -564,15 +567,15 @@ export class StackConverter extends ArtifactConverter implements intrinsics.Intr
 
         switch (target) {
             case 'AWS::AccountId':
-                return getAccountId({ parent: this.app.component }).then((r) => r.accountId);
+                return getAccountId({ parent: this.stackResource }).then((r) => r.accountId);
```

Comment:
The parent should be the stack so that we use the provider associated with the stack.